util/webhook/webhook.go

```diff
@@ -300,6 +300,20 @@ func (a *ArgoCDWebhookHandler) HandleEvent(payload interface{}) {
 			continue
 		}
 		for _, app := range filteredApps {
+			drySource := app.Spec.GetHydratorDrySource()
+			if sourceRevisionHasChanged(drySource, revision, touchedHead) && sourceUsesURL(drySource, webURL, repoRegexp) {
```

Comment:
How about nesting this block in `if app.Spec.SourceHydrator != nil` and then using `app.Spec.SourceHydrator.GetDrySource()`?

Reply:
Thank you for letting me know. I’ve fixed my code.